### PR TITLE
D-F verified language constants

### LINKED ISSUE #308: Add support for D-F wakatime verified languages
Add language constants in `pkg/heartbeat/language.go` for new D-F wakatime verified languages as returned by https://wakatime.com/api/v1/program_languages

### PR DESCRIPTION
**Pull request by gandarez**
This PR ensures that language constants exists for all D-F verified languages, as returned by https://wakatime.com/api/v1/program_languages.

Closes #308

### CODE REVIEW

File: pkg/heartbeat/language.go [hunk 1]
```diff
@@ -1653,19 +1742,29 @@ const (
...
 	languageFontStr                        = "Font"
+	languageFormattedStr                   = "Formatted"
+	languageForthStr                       = "Forth"
 	languageFortranStr                     = "Fortran"
 	languageFortranFixedStr                = "FortranFixed"
+	languageFortranFreeFormStr             = "Fortran Free Form"
 	languageFSharpStr                      = "F#"
 	languageFoxProStr                      = "FoxPro"
 	languageFreefemStr                     = "Freefem"
-	languageFStarStr                       = "FStar"
```
Comment (dron22) [inline pkg/heartbeat/language.go:1668]: `FStar` is used in chroma. As it seems to be valid alternative spelling for `F*`, let's add it to the general aliases (https://github.com/wakatime/wakatime-cli/blob/develop/pkg/heartbeat/language_test.go#L738), as well as to `StringChroma` (https://github.com/wakatime/wakatime-cli/blob/develop/pkg/heartbeat/language_test.go#L939)

**Review by dron22 — CHANGES_REQUESTED** [review]
One remarks regarding `FStar`, where we should be backwards compatible. This will also ensure that it works with chroma.

File: pkg/heartbeat/language_test.go [hunk 2]
```diff
@@ -235,6 +235,7 @@ func languageTests() map[string]heartbeat.Language {
 		"execline":                         heartbeat.LanguageExecline,
 		"Ezhil":                            heartbeat.LanguageEzhil,
 		"F*":                               heartbeat.LanguageFStar,
+		"FStar":                            heartbeat.LanguageFStarLiteral,
```
Comment (dron22) [inline pkg/heartbeat/language_test.go:238]: Here it is not needed. Can be removed.

File: pkg/heartbeat/language_test.go [hunk 3]
```diff
@@ -768,6 +769,7 @@ func languageTestsAliases() map[string]heartbeat.Language {
 		"ApacheConf":        heartbeat.LanguageApacheConfig,
 		"ASP Classic":       heartbeat.LanguageClassicASP,
 		"Batch Script":      heartbeat.LanguageBatchfile,
+		"FStar":             heartbeat.LanguageFStarLiteral,
```
Comment (dron22) [inline pkg/heartbeat/language_test.go:772]: Should resolve to the existing language.
```
"FStar": heartbeat.LanguageFStar,
```

File: pkg/heartbeat/language_test.go [hunk 4]
```diff
@@ -977,6 +979,7 @@ func TestLanguage_StringChroma(t *testing.T) {
 		"EmacsLisp":       heartbeat.LanguageEmacsLisp,
 		"GAS":             heartbeat.LanguageAssembly,
 		"FSharp":          heartbeat.LanguageFSharp,
+		"FStar":           heartbeat.LanguageFStarLiteral,
```
Comment (dron22) [inline pkg/heartbeat/language_test.go:982]: Should resolve to the existing language.
```
"FStar": heartbeat.LanguageFStar,
```

**Review by dron22 — COMMENTED** [review]
We do not need to add a new language. It should resolve to the existing one :+1:

**Comment by gandarez** [comment]
@dron22 ready

File: pkg/heartbeat/language.go [hunk 5]
```diff
@@ -3851,6 +3851,8 @@ func ParseLanguageFromChroma(lexerName string) (Language, bool) {
 		return LanguageGo, true
 	case normalizeString(languageFSharpChromaStr):
 		return LanguageFSharp, true
+	case normalizeString(languageFStarLiteralStr):
```
Comment (dron22) [inline pkg/heartbeat/language.go:3854]: This is not needed, if we handle it in `ParseLanguage`.

File: pkg/heartbeat/language.go [hunk 6]
```diff
@@ -409,32 +455,54 @@ const (
...
 	LanguageFreefem
-	// LanguageFStar represents the FStar programming language.
+	// LangauageFreeMarker represents the FreeMarker programming language.
+	LangauageFreeMarker
+	// LangauageFrege represents the Frege programming language.
+	LangauageFrege
+	// LanguageFStar represents the F* programming language.
 	LanguageFStar
+	// LanguageFStarLiteral represents the FStar programming language.
+	LanguageFStarLiteral
```
Comment (dron22) [inline pkg/heartbeat/language.go:503]: This should be removed completely

**Review by dron22 — CHANGES_REQUESTED** [review]

**Comment by gandarez** [comment]
@dron22 done

**Review by dron22 — APPROVED** [review]
LGTM :+1:

**Pull request merged**